explorer/interpreter/type_checker.cpp

```diff
@@ -5826,26 +5855,27 @@ auto TypeChecker::TypeCheckConstraintTypeDeclaration(
     Nonnull<ConstraintTypeDeclaration*> constraint_decl,
     const ImplScope& impl_scope) -> ErrorOr<Success> {
   if (trace_stream_->is_enabled()) {
-    *trace_stream_ << "** checking ";
-    constraint_decl->PrintID(trace_stream_->stream());
-    *trace_stream_ << "\n";
+    *trace_stream_ << "*** checking `" << PrintAsID(*constraint_decl) << "` ("
+                   << constraint_decl->source_loc() << ")\n";
   }
   ImplScope constraint_scope(&impl_scope);
   if (constraint_decl->params().has_value()) {
     BringPatternImplBindingsIntoScope(*constraint_decl->params(),
                                       constraint_scope);
   }
   if (trace_stream_->is_enabled()) {
-    *trace_stream_ << constraint_scope;
+    *trace_stream_ << "--- impl declarations for `"
+                   << PrintAsID(*constraint_decl) << "` ("
+                   << constraint_decl->source_loc() << ")\n";
+    *trace_stream_ << " +  " << constraint_scope;
```

Comment:
You have a `" +  "` here, and another `" +  "` in the print function for the constraint scope when you need a line continuation. I think it would be better to move all the `" +  "` printing to the same place, in `constraint_scope` itself, and do it before printing the scope rather than before printing the parent. That'd make it easier to keep the formatting consistent. What do you think?

Reply:
Yes it will make it more consistent. Thanks! Doing the change in next commit.